### PR TITLE
chore(testing): add hooks testing library too

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Enhancement**

This PR resolves []

### Description

After adding the react native testing library, it's easy to add the hooks one too.

I added the lib, and moved two hook tests we have to it. One of the tests is a bit funky where it checks the rerenders too, so we still need to use the regular testing inside a component, which is totally fine.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Add hooks testing library - pavlos

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Add hooks testing library - pavlos)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b2625bd491a927afd98809d432776475c72e6e26
</p>

File: package.json [hunk 1]
```diff
@@ -174,8 +174,9 @@
     "@storybook/addon-actions": "^5.3",
     "@storybook/react-native": "^5.3.25",
     "@storybook/react-native-server": "^5.3.23",
-    "@testing-library/jest-native": "^4.0.1",
-    "@testing-library/react-native": "7.1.0",
+    "@testing-library/jest-native": "4.0.1",
+    "@testing-library/react-hooks": "7.0.1",
```
Comment (gkartalis) [inline package.json:178]: 🚀 😍

**Review by gkartalis — APPROVED** [review]
Looks amazing! 😍

**Comment by gkartalis** [comment]
There is a weird error on circle ci on detect-secrets check 🤔 : 
```
#!/bin/bash -eo pipefail
detect-secrets-hook --baseline .secrets.baseline --exclude-secrets '[a-fA-F0-9]{24}' --exclude-lines 'W/"[!#-\x7E]*"' $(git ls-files)
```
The baseline file was updated.
Probably to keep line numbers of secrets up-to-date.
Please `git add .secrets.baseline`, thank you.



Exited with code exit status 3
CircleCI received exit code 3

**Comment by admbtlr** [comment]
This is great! But can you rename it to `chore(testing): add hooks testing library too` please? `enhancement` is not in the spec.

**Review by lordkiz — APPROVED** [review]
LGTM. 
We can merge this after you apply the .secrets.baseline fix

**Comment by pvinis** [comment]
@admbtlr I changed it to `test(hooks)` since it's about tests. I thought we are using the format but not the names? Am I remembering wrong?

**Comment by pvinis** [comment]
Fixing the secrets thing now

**Comment by admbtlr** [comment]
> @admbtlr I changed it to `test(hooks)` since it's about tests. 

`test` usually means that you're adding tests. `chore` is for infrastructure, such as adding libraries - which I think is more relevant in this case.

> I thought we are using the format but not the names? Am I remembering wrong?

I don't remember this ever being decided, but maybe I missed something?

**Review by MounirDhahri — APPROVED** [review]
Nice 🎉

**Comment by pvinis** [comment]
I'll add this to the next KS or mobile practice to talk about this 👍.

**Comment by pvinis** [comment]
What is happening right now?! We need to make this step be in CI. CI can update this file, we shouldn't need to bother.

**Comment by pvinis** [comment]
#squashongreen

**Pull request merged**